### ISSUE TITLE
Attemting to edit newly created task results in error 412 on PUT request (which is silently ignored)

### ISSUE THREAD

**Issue by fordfrog** [owncloud/tasks#330]
### Steps to reproduce
1. create new task
2. click on it to edit
3. edit some attribute, like starting date, ending date or any other
4. watch dev console in browser for related PUT request failing with error 412
5. now if you reload the page the editing works fine

### Expected behaviour
modified task should be saved

### Actual behaviour
due to some mismatch with etag provided in if-match header the request SILENTLY fails. here is the response body content:
```
<?xml version="1.0" encoding="utf-8"?>
<d:error xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns">
  <s:exception>Sabre\DAV\Exception\PreconditionFailed</s:exception>
  <s:message>An If-Match header was specified, but none of the specified the ETags matched.</s:message>
  <s:header>If-Match</s:header>
</d:error>
```

### Server configuration
Gentoo Linux

**Web server:**
Apache 2.4.23

**Database:**
PostgreSQL 9.5.2

**PHP version:**
5.6.23 through fpm

**ownCloud version:** (see ownCloud admin page)
9.0.3

**Tasks version:** (see ownCloud apps page)
0.9.1

**Updated from an older ownCloud or fresh install:**
fresh install

**Signing status (ownCloud 9.0 and above):**
there is only one failing file which is provided by gentoo and does not affect website functionality
```
Results
=======
- core
	- EXTRA_FILE
		- .webapp
```

**List of activated apps:**
```
Enabled:
  - activity: 2.2.1
  - announcementcenter: 1.1.2
  - bookmarks: 0.7
  - calendar: 1.2.2
  - comments: 0.2
  - contacts: 1.3.1.0
  - dav: 0.1.6
  - documents: 0.12.0
  - federatedfilesharing: 0.1.0
  - files: 1.4.4
  - files_pdfviewer: 0.8.1
  - files_sharing: 0.9.1
  - files_texteditor: 2.1
  - files_trashbin: 0.8.0
  - files_versions: 1.2.0
  - files_videoplayer: 0.9.8
  - firstrunwizard: 1.1
  - gallery: 14.5.0
  - mail: 0.5.2
  - music: 0.3.11
  - news: 8.8.0
  - notes: 2.0.1
  - notifications: 0.2.3
  - ownnote: 1.08
  - provisioning_api: 0.4.1
  - shorten: 0.0.15
  - systemtags: 0.2
  - tasks: 0.9.1
  - templateeditor: 0.1
  - updatenotification: 0.1.0
Disabled:
  - encryption
  - external
  - federation
  - files_antivirus
  - files_external
  - galleryplus
  - user_external
  - user_ldap
```

**The content of config/config.php:**
```
{
    "system": {
        "instanceid": "oc99l9y1ozjh",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "<-- REMOVED -->"
        ],
        "datadirectory": "\/var\/www\/<-- REMOVED -->\/data",
        "overwrite.cli.url": "https:\/\/<-- REMOVED -->",
        "dbtype": "pgsql",
        "version": "9.0.3.2",
        "dbname": "owncloud",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "UTC",
        "installed": true,
        "mail_from_address": "owncloud",
        "mail_smtpmode": "php",
        "mail_domain": "startnet.cz",
        "memcache.local": "\\OC\\Memcache\\APCu"
    }
}
```

**Are you using external storage, if yes which one:** local/smb/sftp/...
no

**Are you using encryption:** yes/no
no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
no

### Client configuration
**Browser:**
Chromium 53.0.2785.8 (64-bit)

**Operating system:**
Gentoo Linux

**CalDAV-clients:**
DAVDroid

#### Web server error log
nothing relevant there

#### ownCloud log (data/owncloud.log)
nothing relevant there

#### Browser log
headers from the request and response
```
Request URL:https://<-- REMOVED -->/remote.php/dav/calendars/<-- REMOVED -->/<-- REMOVED -->/ownCloud-x89k3ydjekevet9lqokvb1emi9zykgsyz7omt9sx5dh6t7qfr.ics
Request Method:PUT
Status Code:412 
Remote Address:<-- REMOVED -->:443
Response Headers
cache-control:no-store, no-cache, must-revalidate, post-check=0, pre-check=0
content-encoding:gzip
content-length:227
content-security-policy:default-src 'self'; script-src 'self' 'unsafe-eval'; style-src 'self' 'unsafe-inline'; frame-src *; img-src * data: blob:; font-src 'self' data:; media-src *; connect-src *
content-type:application/xml; charset=utf-8
date:Wed, 20 Jul 2016 14:48:30 GMT
etag:"e3d1ebaa280512ea419ad6ef71fe5879-gzip"
expires:Thu, 19 Nov 1981 08:52:00 GMT
pragma:no-cache
server:Apache
status:412
strict-transport-security:max-age=15768000; includeSubDomains; preload
vary:Accept-Encoding
x-content-type-options:nosniff
x-download-options:noopen
x-frame-options:Sameorigin
x-permitted-cross-domain-policies:none
x-robots-tag:none
x-xss-protection:1; mode=block
Request Headers
:authority:<-- REMOVED -->
:method:PUT
:path:/remote.php/dav/calendars/<-- REMOVED -->/<-- REMOVED -->/ownCloud-x89k3ydjekevet9lqokvb1emi9zykgsyz7omt9sx5dh6t7qfr.ics
:scheme:https
accept:*/*
accept-encoding:gzip, deflate, sdch, br
accept-language:cs,en-US;q=0.8,en;q=0.6,en-GB;q=0.4,it;q=0.2,de;q=0.2,pl;q=0.2,ru;q=0.2,sk;q=0.2,es;q=0.2
cache-control:no-cache
content-length:312
content-type:text/calendar; charset=UTF-8
cookie:oc_sessionPassphrase=<-- REMOVED -->
dnt:1
if-match:"e3d1ebaa280512ea419ad6ef71fe5879-gzip"
origin:<-- REMOVED -->
pragma:no-cache
requesttoken:Gy4wAlRHeDh9FhhhXzN1FhQ3JAQpVipEBw0NVyYoDRY=:jZWPlr2rVU7ThTBbuqejb7G65w44uCfZATK1A+LHQBU=
user-agent:Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/53.0.2785.8 Safari/537.36
Request Payload
BEGIN:VCALENDAR
PRODID:-//ownCloud tasks v0.9.1
BEGIN:VTODO
CREATED:20160720T164812
DTSTAMP:20160720T164812
LAST-MODIFIED:20160720T164812
UID:5w6hg4cuhgwd2ls3986jyk3xr
SUMMARY:test
PRIORITY:0
PERCENT-COMPLETE:0
X-OC-HIDESUBTASKS:0
DTSTART:20160720T170000
DUE:20160720T170000
END:VTODO
END:VCALENDAR
```

**Comment by fordfrog** [comment]
i just found the task also cannot be deleted with the same issue. only after reload of the tasks the newly created task can be modified or deleted. also just tested with oc 9.0.4 which contains the same version of tasks as oc 9.0.3.

**Comment by raimund-schluessler** [comment]
Thanks for the detailed report. Sadly I don't know what causes this. Could you please also provide the the request for the successful edit or delete request (try deleting it directly after creating and after Tasks app reload and report both requests). I want to see what differs in the requests.

**Comment by raimund-schluessler** [comment]
@fordfrog ping.

**Comment by raimund-schluessler** [comment]
@fordfrog Any news on this? I would like to sort it out and will close the issue in the next days if no info provided.

**Comment by fordfrog** [comment]
sorry, i missed your messages. will try to provide you with more information when i get to it.

**Comment by raimund-schluessler** [comment]
I am going to close this issue due to missing information. Please reopen if you are able to provide more information.

**Comment by fordfrog** [comment]
ok, will do that when i get to it. i have really busy schedule. sorry for that.

**Comment by fordfrog** [comment]
@raimund-schluessler how can i provide you with the requests? i saved them from chromium as har files but it's a lot of data to inline it here. also, i would not like to publish them publicly as they are. if so, i'd have to obfuscate some info.
